File: Python/Strings/pg0007.py
#Longest substring without repeating – Find the maximum-length substring that has all unique characters.

s = input()
char_set = set()
left = 0
max_len = 0
start = 0
for right in range(len(s)):
    while s[right] in char_set:
        char_set.remove(s[left])
        left += 1
    char_set.add(s[right])
    
    if right - left + 1 > max_len:
        max_len = right - left + 1
        start = left
print(s[start:start + max_len])

File: Java/Strings/pg0007.java
//Longest substring without repeating – Find the maximum-length substring that has all unique characters.

import java.util.*;

public class pg0007 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();
        int maxLength = 0;
        String longestSubstr = "";
        for (int i = 0; i < s.length(); i++) {
            Set<Character> set = new HashSet<>();
            StringBuilder currentSubstr = new StringBuilder();
            for (int j = i; j < s.length(); j++) {
                char ch = s.charAt(j);
                if (set.contains(ch)) {
                    break;
                }
                set.add(ch);
                currentSubstr.append(ch);
            }
            if (currentSubstr.length() > maxLength) {
                maxLength = currentSubstr.length();
                longestSubstr = currentSubstr.toString();
            }
        }
        System.out.println(longestSubstr);
    }
}
